solidity/contracts/lib/zeto_common.sol

```diff
@@ -24,9 +24,49 @@ import {OwnableUpgradeable} from "@openzeppelin/contracts-upgradeable/access/Own
 /// @author Kaleido, Inc.
 /// @dev Implements common functionalities of Zeto based tokens
 abstract contract ZetoCommon is IZeto, OwnableUpgradeable {
-    function __ZetoCommon_init(address initialOwner) internal onlyInitializing {
+    string private _name;
+    string private _symbol;
+
+    function __ZetoCommon_init(
+        string memory name_,
+        string memory symbol_,
+        address initialOwner
+    ) internal onlyInitializing {
         __Ownable_init(initialOwner);
+        _name = name_;
+        _symbol = symbol_;
+    }
+    /**
+     * @dev Returns the name of the token.
+     */
+    function name() public view virtual returns (string memory) {
+        return _name;
+    }
+
+    /**
+     * @dev Returns the symbol of the token, usually a shorter version of the
+     * name.
+     */
+    function symbol() public view virtual returns (string memory) {
+        return _symbol;
     }
+
+    /**
+     * @dev Returns the number of decimals used to get its user representation.
+     * For example, if `decimals` equals `2`, a balance of `505` tokens should
+     * be displayed to a user as `5.05` (`505 / 10 ** 2`).
+     *
+     * Tokens usually opt for a value of 18, imitating the relationship between
+     * Ether and Wei. But Zeto uses 2 as the default, based on its target use cases
+     * in CBDCs and tokenized commercial money. The default value can be overridden.
```

Comment:
I agree 18 always seems like overkill, so I'm OK with a lower default. 2 seems a bit too low though, as I think fractional cents are common in these use cases, especially for interest calculations, etc.

Perhaps we should pick a higher default like 4 (common for interest rates and FX) or 8 (default for BTC, or areas that involve very small fractional shares). As you note, it is only for display - but it will strongly frame how these use cases are built on top of Zeto.

Reply:
great feedback on the use cases. maybe 4 for now given the focus on enterprise use cases. the sample in `contracts/test/tendecimals.sol` also demonstrated how easy it is to override that.